rotkehlchen/exchanges/coinbaseprime.py

```diff
@@ -0,0 +1,477 @@
+import base64
+import hashlib
+import hmac
+import logging
+from collections import defaultdict
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any, Final, Literal, overload
+from urllib.parse import urlparse
+
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.converters import asset_from_coinbase
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.asset import UnknownAsset, UnsupportedAsset
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.exchanges.data_structures import AssetMovement, Fee, MarginPosition, Trade
+from rotkehlchen.exchanges.exchange import ExchangeInterface, ExchangeQueryBalances
+from rotkehlchen.inquirer import Inquirer, Price
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_asset_amount_force_positive,
+    deserialize_fee,
+    deserialize_fval,
+)
+from rotkehlchen.types import (
+    ApiKey,
+    ApiSecret,
+    AssetMovementCategory,
+    ExchangeAuthCredentials,
+    Location,
+    Timestamp,
+    TradeType,
+)
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.misc import iso8601ts_to_timestamp, timestamp_to_iso8601, ts_now
+from rotkehlchen.utils.mixins.cacheable import cache_response_timewise
+from rotkehlchen.utils.mixins.lockable import protect_with_lock
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import AssetWithOracles
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.history.events.structures.base import HistoryEvent
+
+
+PRIME_BASE_URL: Final = 'https://api.prime.coinbase.com/v1'
+COMPLETED_TRANSACTION_STATUS: Final = {
+    'TRANSACTION_DONE',
+    'TRANSACTION_IMPORTED',
+}
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _process_trade(trade_data: dict[str, Any]) -> Trade | None:
+    """Process trade from coinbase prime. Returns None if the order can't be processed
+    May raise:
+    - DeserializationError
+    """
+    try:
+        if trade_data['status'] != 'FILLED':
+            return None
+
+        if type(trade_data['product_id']) is not str and '-' not in trade_data['product_id']:
```

Comment:
Don't check type with `type`. Always `isinstance`. Various benefits such as also subclasses etc. Google or ask claude for details

```suggestion
        if not isinstance(trade_data['product_id'], str) and '-' not in trade_data['product_id']:
```